### PR TITLE
Supply count for map_accounts, map_roles and map_users

### PR DESCRIPTION
**Pull request by Chili-Man**
# PR o'clock

## Description

We have a wrapper module around this (to enforce certain settings when invoked) and as such,  we're having issues passing in `map_accounts`, `map_roles`, `map_users` because of the `can not compute value of 'count'`. It pains me to have to do this, but this PR adds in count variables to be explicitly passed in for `map_accounts`,  `map_roles` and `map_users`  as a workaround for Terraforms limitations with respect to computed values for `count`  

### Checklist

- [X] `terraform fmt` and `terraform validate` both work from the root and `examples/eks_test_fixture` directories (look in CI for an example)
- [X] Tests for the changes have been added and passing (for bug fixes/features)
- [X] Test results are pasted in this  PR (in lieu of CI) 
[new-cluster-tf-plan.txt](https://github.com/terraform-aws-modules/terraform-aws-eks/files/2655806/new-cluster-tf-plan.txt)
[apply-output.txt](https://github.com/terraform-aws-modules/terraform-aws-eks/files/2655807/apply-output.txt)

- [X] Docs have been updated using `terraform-docs` per `README.md` instructions
- [X] I've added my change to CHANGELOG.md
- [x] Any breaking changes are highlighted above

### CODE REVIEW

**Comment by Chili-Man** [comment]
The Travis CI test is currently failing for an unrelated error:
```
The command "unzip terraform.zip ; rm -f terraform.zip; chmod +x terraform" failed and exited with 1 during .
```

**Comment by max-rocket-internet** [comment]
@antonbabenko what are other modules doing about issues like this?

It's the classic `can not compute value of 'count'` problem.

We currently have 2 unmerged PRs to solve the same type of problem, this one and https://github.com/terraform-aws-modules/terraform-aws-eks/pull/186

I know they solve a problem but they also make the code messy and harder to read.

Thoughts?

**Comment by RothAndrew** [comment]
> @antonbabenko what are other modules doing about issues like this?
> 
> It's the classic `can not compute value of 'count'` problem.
> 
> We currently have 2 unmerged PRs to solve the same type of problem, this one and #186
> 
> I know they solve a problem but they also make the code messy and harder to read.
> 
> Thoughts?

This is usually how it is done. It's ugly, but it works.

**Comment by antonbabenko** [comment]
Yes, this is how I would deal with this problem here also. Ugliness in the module is fine as long as users of the module know how to use it and which values to specify.

**Comment by RothAndrew** [comment]
I would definitely make this is a "major" version increment, since it isn't backward compatible for those variables.

**Comment by Chili-Man** [comment]
@max-rocket-internet  I've made it a major bump per @RothAndrew's feedback and I also added a count variable for the `map_accounts` for consistency

File: README.md [hunk 1]
```diff
@@ -101,46 +101,50 @@ MIT Licensed. See [LICENSE](https://github.com/terraform-aws-modules/terraform-a
...
+| manage_aws_auth | Whether to write and apply the aws-auth configmap file. | string | `true` | no |
+| map_accounts | Additional AWS account numbers to add to the aws-auth configmap. See examples/eks_test_fixture/variables.tf for example format. | list | `<list>` | no |
+| map_accounts_count | The count of accounts in the map_accounts list. | string | `0` | no |
+| map_roles | Additional IAM roles to add to the aws-auth configmap. See examples/eks_test_fixture/variables.tf for example format. | list | `<list>` | no |
+| map_roles_count | The count of roles in the map_roles list. | string | `0` | no |
+| map_users | Additional IAM users to add to the aws-auth configmap. See examples/eks_test_fixture/variables.tf for example format. | list | `<list>` | no |
+| map_users_count | The count of roles in the map_users list. | string | `0` | no |
 | subnets | A list of subnets to place the EKS cluster and workers within. | list | - | yes |
 | tags | A map of tags to add to all resources. | map | `<map>` | no |
-| vpc\_id | VPC where the cluster and workers will be deployed. | string | - | yes |
```
Comment (max-rocket-internet) [inline README.md]: Sorry but could you update to the later version of `terraform-docs` in order to keep the new format with underscores?

File: examples/eks_test_fixture/variables.tf [hunk 2]
```diff
@@ -42,3 +54,9 @@ variable "map_users" {
     },
   ]
 }
+
+variable "map_users_count" {
+  description = "The count of roles in the map_users list."
+  type        = "string"
+  default     = 1
```
Comment (max-rocket-internet) [inline examples/eks_test_fixture/variables.tf]: Shouldn't this be 2?

File: examples/eks_test_fixture/variables.tf [hunk 3]
```diff
@@ -12,6 +12,12 @@ variable "map_accounts" {
   ]
 }
 
+variable "map_accounts_count" {
+  description = "The count of accounts in the map_accounts list."
+  type        = "string"
+  default     = 1
```
Comment (max-rocket-internet) [inline examples/eks_test_fixture/variables.tf]: Shouldn't this be 2?

**Comment by max-rocket-internet** [comment]
Thank you @Chili-Man (funny username haha)

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,14 +5,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) and this
 project adheres to [Semantic Versioning](http://semver.org/).
 
-## [[v1.9.0](https://github.com/terraform-aws-modules/terraform-aws-eks/compare/v1.8.0...HEAD)] - 2019-01-??]
 
-### Added
+## [[v2.0.0](https://github.com/terraform-aws-modules/terraform-aws-eks/compare/v1.8.0...HEAD)] - 2019-01-??]
 
+### Added
+- (Breaking Change) New input variables `map_accounts_count`, `map_roles_count` and `map_users_count` to allow using computed values as part of `map_accounts`, `map_roles` and `map_users` configs (by @chili-man on behalf of OpenGov).
```
Comment (chenrui333) [inline CHANGELOG.md]: I realized this is very important message as recently I upgrade my cluster but run into the issue with users in the map_users did not get provisioned properly.

**Pull request merged**